pytest/tests/sanity/congestion_control_bootstrap.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+import unittest
+import pathlib
+import sys
+import time
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+
+from configured_logger import logger
+import cluster
+import utils
+
+EPOCH_LENGTH = 5
+NUM_SHARDS = 4
+
+
+class CongestionControlBootstrapTest(unittest.TestCase):
+    """Tests congestion control bootstrapping in the case of node restart.
+     
+    It runs the chain multiple epochs so that garbage collection kicks in.
+    Then it restarts the nodes to ensure that congestion control bootstrapping does not fail."""
+
+    def test(self):
+        self.start_nodes()
+        self.check_congestion_info()
+        # Now wait until there are enough epochs to pass
+        # for GC to kick in and then restart the nodes.
+        self.wait_for_multiple_epochs()
+        self.restart_nodes()
+        self.check_congestion_info()
+
+    def start_nodes(self):
+        self.nodes = cluster.start_cluster(
+            num_nodes=4,
+            num_observers=0,
+            num_shards=NUM_SHARDS,
+            config=None,
+            genesis_config_changes=[["epoch_length", EPOCH_LENGTH],
+                                    ["block_producer_kickout_threshold", 0],
+                                    ["chunk_producer_kickout_threshold", 0]],
+            client_config_changes={
+                i: {
+                    "tracked_shards": [0],
+                    "gc_num_epochs_to_keep": 2
+                } for i in range(4)
+            })
+        utils.wait_for_blocks(self.nodes[0], count=3)
+
+    def wait_for_multiple_epochs(self):
+        utils.wait_for_blocks(self.nodes[0], count=5 * EPOCH_LENGTH)
+
+    def restart_nodes(self):
+        for i in range(len(self.nodes)):
+            self.nodes[i].kill()
+            time.sleep(2)
+            self.nodes[i].start(boot_node=None if i == 0 else self.nodes[0])
```

Comment:
I see you kill them one at a time. Does the chain keep going since you have 3/4 = 75% >= 66% of stake online at any time?

Reply:
I actually did not give attention to what happens to the chain during/after the restart, as this tests if the nodes can restart. This could be done just stopping all nodes before restarting.